src/index.ts

```diff
@@ -0,0 +1,154 @@
+ * @todo Rework the url checker and add documentation
+ */
+class Client implements IClient {
+  constructor(ep: Endpoint) {
+    /*
+     * When no protocol is defined, http will be appended. Therefore "https"
+     * should always be included for performance.
+     */
+    ep.url = ((url: string) => {
+      if (!/^(?:f|ht)tps?\:\/\//.test(url)) {
+        url = "http://" + url;
+      }
+
+      return url;
+    })(ep.url);
+  }
+}
+
+/** @class A client implementation for SNEK interaction */
+class SnekClient extends Client {
+  //#LEGACY
+  /** @deprecated Will be removed in the upcoming release */
+  endpoint: ApolloEndpoint;
+  gql: ApolloEndpoint;
+  template: IMainTemplate;
+  session: SnekSession;
+
+  /**
+   * @constructor
+   * @author Nico Schett <contact@schett.net>
+   * @param url The base URL the SnekClient should be working on.
+   *            Default: "https://engine.snek.at/api/graphiql".
+   * @param headers A object containing various request headers
+   * @param type A type description to differ between multiple instances
+   */
+  constructor(
+    url: string = "https://engine.snek.at/api/graphiql",
+    headers: object = {},
+    type: string = "graphql"
+  ) {
+    super({ type, url, headers });
+
+    this.template = new MainTemplate();
+    this.gql = new Apollo(url, { headers });
+    this.endpoint = this.gql;
+    this.session = new SnekSession("snek", this.gql, this.template.snek);
+  }
+}
+
+/** @class A client implementation for github interaction */
+class GithubClient extends Client {
+  //#LEGACY
+  /** @deprecated Will be removed in the upcoming release */
+  endpoint: ApolloEndpoint;
+  gql: ApolloEndpoint;
+  template: IMainTemplate;
+  session: GithubSession;
+
+  /**
+   * @constructor
+   * @author Nico Schett <contact@schett.net>
+   * @param url The base URL the GithubClient should be working on.
+   *            Default: "https://api.github.com/graphql".
+   * @param headers A object containing various request headers
+   * @param type A type description to differ between multiple instances
+   */
+  constructor(
+    url: string = "https://api.github.com/graphql",
+    headers: object = {},
+    type: string = "graphql"
+  ) {
+    super({ type, url, headers });
+
+    this.template = new MainTemplate();
+    this.gql = new Apollo(url, { headers });
+    this.endpoint = this.gql;
+    this.session = new GithubSession("github", this.gql, this.template);
+  }
+}
+
+/**
+ * @class A client implementation for web interaction like scraping or access
+ *        to APIv4 endpoints.
+ */
+class WebClient extends Client {
+  public scraper: ScraperEndpoint;
+
+  /**
+   * @constructor
+   * @author Nico Schett <contact@schett.net>
+   * @param url The base URL the WebClient should be working on
+   * @param headers A object containing various request headers
+   * @param type A type description to differ between multiple instances
+   */
```

Comment:
Should have some sort of comment.

Initializes a Web client.

Reply:
Yeah, but not your suggestion cause this includes too less/useless information.
We have to discuss about a new TONIC rule in snek-at/tonic/issues/3